cobc/parser.y

```diff
@@ -12900,15 +12900,17 @@ close_body:
 close_files:
   file_name _close_option
   {
-#if 0 /* CHECKME: likely not needed */
-	begin_implicit_statement ();
-#endif
-	cb_emit_close ($1, $2);
+	  struct cb_statement * saved_current_statement = current_statement ;
+	  begin_implicit_statement ();
```

Comment:
I removed the modification of `delete_file_list`. The reason is that the problem appears with `DECLARATIVES`, and it looks like `DELETE FILE` does not trigger them in case of error (in the C code generated, there is no test after `cob_delete_file()` to check for an exception condition). So, we could use the same approach to construct the tree, but it would not change the generated code at the end.

Reply:
I've rechecked the standard. While it is not completely clear what to do for `DELETE RECORD`, it is clear for `DELETE FILE` - we _should_ go into `DECLARATIVES` for any exception.

Can you please fix this along in this PR?

Reply:
This one is the "open" part, as it is only "related", you may commit to svn as-is, but as it _should_ work it would be reasonable to at least do the same change to delete_file that you did to the others.